compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -1377,7 +1391,10 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
 
                 let largest_niche = Niche::from_scalar(dl, Size::ZERO, tag);
 
-                let tagged_layout = Layout {
+                let layout_variants =
+                    layout_variants.into_iter().map(|v| tcx.intern_layout(v)).collect();
```

Comment:
hmm, i wonder if we could intern them beforehand to avoid another loop? the perf didn't show any changes so I don't know about the impact of this code.